unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -0,0 +1,310 @@
+
+            // var requestEvents = await rpc.Social().GetRequestEvents(new Empty());
+            //
+            // foreach (var friendRequest in requestEvents.Incoming.Items)
+            // {
+            //     OnIncomingFriendRequestAdded?.Invoke(new FriendRequest(
+            //         GetFriendRequestId(friendRequest.User.Address, friendRequest.CreatedAt), friendRequest.CreatedAt, friendRequest.User.Address, ownUserProfile.userId, friendRequest.Message));
+            // }
+            //
+            // foreach (var friendRequest in requestEvents.Outgoing.Items)
+            // {
+            //     OnOutgoingFriendRequestAdded?.Invoke(new FriendRequest(
+            //         GetFriendRequestId(friendRequest.User.Address, friendRequest.CreatedAt), friendRequest.CreatedAt, ownUserProfile.userId, friendRequest.User.Address, friendRequest.Message));
+            // }
+
+            await UniTask.SwitchToMainThread(cancellationToken);
+
+            return new FriendshipInitializationMessage()
+            {
+                // totalReceivedRequests = requestEvents.Incoming.Items.Count,
+                totalReceivedRequests = 0
+            };
+        }
+
+        public async UniTask ListenToFriendEvents(CancellationToken cancellationToken = default)
+        {
+            var ownUserProfile = userProfileWebInterfaceBridge.GetOwn();
+
+            await UniTask.WaitUntil(() => ownUserProfile.userId != null, PlayerLoopTiming.Update, cancellationToken);
+
+            var stream = socialClient.SubscribeFriendshipEventsUpdates(new Payload() { SynapseToken = "" });
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            await foreach (var item in stream.WithCancellation(cancellationToken))
+            {
+                foreach (var friendshipEvent in item.Events)
+                {
+                    // var action = ToFriendshipAction(friendshipEvent.BodyCase);
+                    //
+                    // OnReceivedFriendshipEvent?.Invoke(new FriendshipUpdateStatusMessage()
+                    // {
+                    //     case FriendshipEventResponse.BodyOneofCase.None: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Request: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Accept: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Reject: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Delete: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Cancel: break;
+                    //     default: throw new ArgumentOutOfRangeException();
+                    // }
+                }
+            }
+        }
+
+        public async UniTask RejectFriendship(string friendRequestId, CancellationToken cancellationToken = default)
+        {
+            string userId = GetUserIdFromFriendRequestId(friendRequestId);
```

Comment:
a good idea would be to check if you have a valid access token (`!string.IsNullOrEmpty(accessToken)`) before executing these kind of operations. In case the accesToken is null, then throw an exception